### ISSUE TITLE
Build error with undefined reference. (about libgupnp)

### ISSUE THREAD

**Issue by shouhin0077** [cablelabs/rui-http-server#1]
I tried to build rui-http-server by reference to README.md in Ubuntu 14.04. Building process was failed with "undefined reference" as follows.
How do I link to libgupnp-1.0.so using tup? libgupnp-1.0.so is installed in /usr/lib by "apt-get install libgupnp-1.0-dev".

```
$ tup upd
[ tup ] [0.010s] Scanning filesystem...
[ tup ] [0.025s] Reading in new environment variables...
[ tup ] [0.039s] Parsing Tupfiles...
 1) [0.001s] .
 2) [0.001s] deps                                                              
 3) [0.001s] deps/librui-discoverer                                            
 4) [0.004s] deps/librui-discoverer/pkgconfig                                  
 5) [0.001s] deps/librui-discoverer/src                                        
 6) [0.001s] deps/librui-discoverer/scripts                                    
 7) [0.001s] deps/librui-discoverer/vapi                                       
 8) [0.001s] src                                                               
 9) [0.001s] vapi                                                              
 10) [0.002s] static                                                           
 11) [0.001s] static/js                                                        
 12) [0.001s] static/images                                                    
 13) [0.000s] static/css                                                       
 [  ETA~=<1s   ] 100%
[ tup ] [0.132s] No files to delete.                                           
[ tup ] [0.132s] Generating .gitignore files...
[ tup ] [0.135s] Executing Commands...
 1) [0.096s] deps/librui-discoverer/pkgconfig: ../scripts/gen-pkgconfig.py     
 2) [0.578s] deps/librui-discoverer/src: valac discoverer.vala --vapidir=../vapi --pkg=gupnp-1.0 --pkg=gee-0.8 -X -w -g --vapi=rui-discoverer-1.0.vapi --library rui-discoverer -H rui-discoverer.h -X -fPIC -X -c -o librui-discoverer.o
 3) [0.024s] deps/librui-discoverer/src: gcc -fPIC -o librui-discoverer.so librui-discoverer.o -shared -fPIC
* 4) src: valac main.vala server.vala --vapidir=../vapi --pkg=gupnp-1.0 --pkg=gee-0.8 --pkg=json-glib-1.0 --pkg=gio-2.0 --pkg=posix -X -w -g --vapidir=./../deps/librui-discoverer/src --includedir=./../deps/librui-discoverer/src -X -I./../deps/librui-discoverer/src -X ./../deps/librui-discoverer/src/librui-discoverer.o --pkg=rui-discoverer-1.0 -o server
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_handle_compatible_uis':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:53: undefined reference to `gupnp_service_info_get_url_base'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:56: undefined reference to `gupnp_service_proxy_end_action'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:64: undefined reference to `gupnp_service_info_get_udn'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:64: undefined reference to `gupnp_service_info_get_udn'
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_handle_ui_listing_update':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:76: undefined reference to `gupnp_service_proxy_begin_action'
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_service_proxy_available':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:85: undefined reference to `gupnp_service_proxy_set_subscribed'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:87: undefined reference to `gupnp_service_proxy_add_notify'
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_subscription_lost':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:94: undefined reference to `gupnp_service_info_get_udn'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:96: undefined reference to `gupnp_service_info_get_udn'
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_service_proxy_unavailable':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:103: undefined reference to `gupnp_service_info_get_udn'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:105: undefined reference to `gupnp_service_info_get_udn'
./../deps/librui-discoverer/src/librui-discoverer.o: In function `rui_discoverer_start':
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:110: undefined reference to `gupnp_context_new'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:112: undefined reference to `gupnp_control_point_new'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:117: undefined reference to `gssdp_resource_browser_set_active'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:119: undefined reference to `gssdp_client_get_host_ip'
/home/iddk/temp/rui-http-server/.tup/mnt/@tupjob-67/home/iddk/temp/rui-http-server/deps/librui-discoverer/src/discoverer.vala:119: undefined reference to `gupnp_context_get_port'
collect2: error: ld returned 1 exit status
error: cc exited with status 256
Compilation failed: 1 error(s), 0 warning(s)
 *** tup errors ***
 *** Command ID=79 failed with return value 1
tup error: Expected to write to file 'server' from cmd 79 but didn't
 *** Additionally, command 79 failed to process input dependencies. These should probably be fixed before addressing the command failure.
 [    ] 100%
 *** tup: 1 job failed.
```

**Comment by brendanlong** [comment]
Hm.. The GUPnP bindings should be bundled since Ubuntu's bindings are wrong. I'll take a look. Did you run this command from the README:

    sudo apt-get install git valac libgupnp-1.0-dev libgee-0.8-dev libjson-glib-dev

?

**Comment by brendanlong** [comment]
Hm I'm seeing this problem in an Ubuntu 14.04 VM too. I'm not really sure what's wrong, since gupnp-1.0 is installed..

**Comment by brendanlong** [comment]
This should be fixed in commit f60cfe084a2b5a0a510e1b7be931800886a8e154. Thanks for the bug report.
